navigator_helpers/tasks/sample_to_dataset/pipeline.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import math
+import time
+
+from typing import Tuple
+
+import pandas as pd
+
+from navigator_helpers.llms.base import LLMRegistry
+from navigator_helpers.logs import get_logger, SIMPLE_LOG_FORMAT
+from navigator_helpers.tasks.sample_to_dataset.task_suite import (
+    SampleToDatasetConfig,
+    SampleToDatasetTaskSuite,
+)
+
+logger = get_logger(__name__, fmt=SIMPLE_LOG_FORMAT)
+
+
+class Sample2DataSetPipeline:
+    """
+    A pipeline for generating synthetic datasets based on a sample dataset.
+
+    This class orchestrates the process of generating a larger dataset from a small sample,
+    using various LLM-powered tasks to create diverse and representative data.
+    """
+
+    def __init__(
+        self, config: SampleToDatasetConfig, llm_registry: LLMRegistry
+    ) -> None:
+        """
+        Initialize the Sample2DataSetPipeline.
+
+        Args:
+            config (SampleToDatasetConfig): Configuration for the sample-to-dataset task.
+            llm_registry (LLMRegistry): Registry of Language Model interfaces.
+        """
+        self.config = config
+        self.llm_registry = llm_registry
+        self.tasks = SampleToDatasetTaskSuite(
+            config=config, llm_registry=llm_registry, logger=logger
+        )
+        self._setup_logging()
+
+    def _setup_logging(self) -> None:
+        """
+        Set up logging configuration to suppress unwanted logs.
+        """
+        for module in ["groq", "mistralai", "openai", "httpx", "httpcore", "autogen"]:
+            logging.getLogger(module).setLevel(logging.WARNING)
+
+    def run(
+        self,
+        sample_dataset: pd.DataFrame,
+        dataset_context: str = "",
+        num_records: int = 20,
+        num_records_per_seed: int = 5,
+        max_workers: int = 4,
+        system_prompt_type: str = "cognition",
```

Comment:
nit: this prompt type is passed through a few layers and it's a bit hard to know what the allowed values are from here (you need to go all the way to where it's used, where we check `if prompt_type not in ["reflection", "cognition"]:`).
Few options:
- have an enum of allowed prompt types
- make this type to be `Literal["cognition","reflection"]`

However, both would need to be passed through all the layers, so we may leave it for now.

Reply:
I created an enum in Jarvis!

Reply:
👏

Reply:
Keeping as is for now, given the TBD of the enum